cli/kata-check.go

```diff
@@ -454,3 +459,45 @@ func genericCheckKVMExtensions(extensions map[string]kvmExtension) (map[string]i
 
 	return results, nil
 }
+
+// func checkVersionConsistencyInComponents checks version consistency in kata components.
+func checkVersionConsistencyInComponents(c *cli.Context) error {
+	config, ok := c.App.Metadata["runtimeConfig"].(oci.RuntimeConfig)
+	if !ok {
+		return errors.New("cannot determine runtime config")
+	}
+
+	proxyInfo, _ := getProxyInfo(config)
+
+	shimInfo, err := getShimInfo(config)
+	if err != nil {
+		return err
+	}
+
+	var runtimeVersion VersionInfo
+	versions := strings.Split(version, ".")
+	if len(versions) > 1 {
+		runtimeVersion.Major = versions[0]
+		runtimeVersion.Minor = versions[1]
+	} else {
+		return errors.New("getting kata-runtime version failed")
+	}
+
+	if proxyInfo.Version.Version != "" {
+		if !versionCompare(proxyInfo.Version, runtimeVersion) || !versionCompare(shimInfo.Version, runtimeVersion) {
+			fmt.Printf("Warning: there exists version inconsistency in kata components. kata-proxy: v%s.%s, kata-shim: v%s.%s, kata-runtime: v%s.%s\n", proxyInfo.Version.Major, proxyInfo.Version.Minor,
```

Comment:
- s/kata components/Kata components/
- We use semver consistently so I think we should display all three parts of the version here (major, minor + patch) , or just the entire version string with a single `%s`.
- Random thought: this would expand the scope of this PR a little, but I wonder if we want to add a `--strict` option so that if you run `kata-runtime kata-check --strict` and one of these checks fails, it returns `1` rather than `0`. That could be more useful for scripts, *etc* which shouldn't have to parse `stderr`. That would imply this function would then return an actual error which itself would make it easier to unit test.